cluster/cluster_members.go

```diff
@@ -86,10 +86,71 @@ func (c InternalClusterMember) ToAPI() (*internalTypes.ClusterMember, error) {
 	}, nil
 }
 
+// prepareUpdateV1 creates the temporary table `internal_cluster_members_new` if we have not yet run `updateFromV1`.
+// To keep this table in sync with `internal_cluster_members`, a create & update trigger is created as well.
+// This table (and its triggers) will be deleted by `updateFromV1`.
+func prepareUpdateV1(tx *sql.Tx) (tableName string, err error) {
+	// Check if we need a temporary table, if no `type` field exists in the schemas table.
+	stmt := "SELECT count(name) FROM pragma_table_info('schemas') WHERE name IN ('type');"
+	var count int
+	err = tx.QueryRow(stmt).Scan(&count)
+	if err != nil {
+		return "", err
+	}
+
+	// If we have a `type` field, then the default database is valid, so we can just use that.
+	// TODO: Dynamically rename to "core_"
+	if count == 1 {
+		return "internal_cluster_members", nil
+	}
+
+	// Check if another cluster member has already created the temporary table.
+	stmt = "SELECT name FROM sqlite_master WHERE name = 'internal_cluster_members_new';"
+	err = tx.QueryRow(stmt).Scan(&tableName)
+	if err != nil && err != sql.ErrNoRows {
+		return "", err
+	}
+
+	if tableName != "" {
+		return tableName, nil
+	}
+
+	// If no cluster member has created the temporary table, create it and return its name.
+	// Also create some triggers so that updates to the `internal_cluster_members` table are propagated to the new table.
+	stmt = `
+CREATE TABLE internal_cluster_members_new (
+  id                   INTEGER   PRIMARY  KEY    AUTOINCREMENT  NOT  NULL,
+  name                 TEXT      NOT      NULL,
+  address              TEXT      NOT      NULL,
+  certificate          TEXT      NOT      NULL,
+  schema_internal      INTEGER   NOT      NULL,
+  schema_external      INTEGER   NOT      NULL,
+  heartbeat            DATETIME  NOT      NULL,
+  role                 TEXT      NOT      NULL,
+  UNIQUE(name),
+  UNIQUE(certificate)
+);
+
+INSERT INTO internal_cluster_members_new SELECT id,name,address,certificate,1,(schema-1),heartbeat,role FROM internal_cluster_members;
```

Comment:
Right, prior to `updateFromV1`, the internal schema version is 1 (because we only have `updateFromV0`). This if the schema version is `>1`, those are all external updates.

Reply:
Thanks.